notebooks/GWAS_blood_pressure_p3/main.py

```diff
@@ -0,0 +1,647 @@
+with herzog.Cell("markdown"):
+    """
+    # Generate a genetic relatedness matrix (GRM)
+    Hail has built-in functions for generating a GRM. A GRM can also account for population stratification and cryptic genetic relatedness in our cohort.
+    """
+# On chr 1 from Amish this only took 46 seconds but on all Amish chromosomes it seemed to have hung
+with herzog.Cell("python"):
+    # Calculate the GRM
+    # WARNING: This can take a very long time to complete!
+    start_grm_time = time.time()
+
+    grm = hl.genetic_relatedness_matrix(mt.GT).to_numpy()
+
+    elapsed_grm_time = time.time() - start_grm_time
+    print(timedelta(seconds=elapsed_grm_time))
+with herzog.Cell("python"):
+    # Get the right sample order
+    ind_order = mt.s.collect()
+with herzog.Cell("python"):
+    # GRM to data frame
+    rel_pd = pd.DataFrame(data=grm,
+                          index=ind_order,
+                          columns=ind_order)
+with herzog.Cell("python"):
+    # Export GRM
+    rel_pd.to_csv(kinship_out)
+with herzog.Cell("markdown"):
+    """
+    # Save sample metadata and update data table
+    Now that we've explored the phenotypes and generated relatedness measures, the next step is to save and push our results back to the workspace data model. Once we populate the data model, we can use it for downstream analyses.
+
+    ## Convert the phenotype data to the correct format
+    """
+with herzog.Cell("python"):
+    # Gather the sample metadata
+    samples_traits_for_analysis = (
+        mt.cols()
+    )
+with herzog.Cell("python"):
+    # Select only the columns we want to keep
+    samples_traits_for_analysis = (
+        samples_traits_for_analysis
+        .key_by()
+        .select('s', 'pheno')
+    )
+with herzog.Cell("python"):
+    # Convert the Hail matrix back to a pandas dataframe
+    samples_traits_for_analysis = (
+        samples_traits_for_analysis
+        .to_pandas()
+    )
+with herzog.Cell("python"):
+    # Change the column names of our data
+    # You can see what column headers are required for the Genesis workflows (for example, sex)
+    col_map = {'s': 'nwd_id',
+               'pheno.subject_id': 'subject_id',
+               'pheno.blood_pressure_test_age_at_bp_systolic': 'age_at_bp_systolic',
+               'pheno.demographic_annotated_sex': 'sex',
+               'pheno.medication_antihypertensive_meds': 'antihypertensive_meds',
+               'pheno.blood_pressure_test_bp_diastolic': 'bp_diastolic',
+               'pheno.blood_pressure_test_bp_systolic': 'bp_systolic',
+               }
+    samples_traits_for_analysis.rename(columns=col_map, inplace=True)
+with herzog.Cell("python"):
+    # Check that this work
+    # Note that because we used the nwd_id as the key to match the phenotypic to genotypic data,
+    # nwd_id is now the firt column
+    samples_traits_for_analysis.head()
+with herzog.Cell("markdown"):
+    """
+    ## Write the phenotype data to a new file
```

Comment:
We should explain the difference between the bp-phenotypes.csv generated from the first notebook and the new phenotype file you create in this second notebook. Something like:

## Write the phenotype data exported from the outputs of Hail to a new file for use in the GENESIS workflows. Hail may have removed individuals if they did not have genotype data associated with them.